cpp-ch/local-engine/Parser/LocalExecutor.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <Common/BlockIterator.h>
+#include <QueryPipeline/QueryPipelineBuilder.h>
+#include <Processors/Executors/PullingPipelineExecutor.h>
+#include <Processors/QueryPlan/QueryPlan.h>
+#include <Parser/CHColumnToSparkRow.h>
+#include <Parser/RelMetric.h>
+
+namespace local_engine
+{
+
+struct SparkBuffer
+{
+    char * address;
+    size_t size;
+};
+
+class LocalExecutor : public BlockIterator
+{
+public:
+    static LocalExecutor * getCurrentExecutor() { return current_executor; }
```

Comment:
Worry about making this a global variable? Could explain it lifetime?

Reply:
add comment